src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
> This is not the signature I suggested

I'm sorry, I misunderstood, so you're still arguing in favor of an overload like this?

```php
interface StreamFactoryInterface {
    public function createStream($resource_or_content);
}
```

I am very much against overloading in this way, as these overloads are, in my opinion, for two distinct use-cases: one in which I *have* a `resource`, and another in which there *is* no resource and I'm instead creating and populating a *new* resource with content.

I don't even know how to name such an interface/method, as it's sometimes a "temp stream factory" and other times a "file stream factory" - I expect that's how this interface/method ends up with a very non-descript name.

The temp stream use-case would make more sense to me as:

```php
interface TempStreamFactoryInterface {
    public function createTempStream($content = "");
}
```

This is in fact also an overload, but a more reasonable overload - one in which you create an empty stream, and another in which you pre-populate a empty stream with data; but in both cases creating a `php://temp` stream, so that seems a more natural overload.

At the moment, this is what I'm proposing:

```php
FileStreamFactory::createFileStream(string $path, string $mode = "r");
ResourceStreamFactory::createResourceStream(resource $handle);
TempStreamFactory::createTempStream(string $content = "");
```

Each creates a distinct, consistent type of Stream from a distinct set of values, and, in my opinion, each has a distinct set of relevant, unambiguous use-cases, and the interface/method/parameter-names are self-describing.

I do not see the benefit of translating a simple, self-explanatory set of interfaces and methods into a singular all-purpose stream factory interface along with a bunch of documentation and implementation requirements in english, when it's possible to create a much more formal and self-explanatory definition.

Anyhow, if no one else agrees or shares my concerns, I have nothing further to add.